src/util/et/error_message.c

```diff
@@ -278,24 +278,32 @@ add_error_table(const struct error_table *et)
 errcode_t KRB5_CALLCONV
 remove_error_table(const struct error_table *et)
 {
-    struct et_list **ep, *e;
+    struct et_list *epp, *ep;
 
+    /* be safe */
+    if (et_list == NULL || terminated)
+        return ENOENT;
     if (CALL_INIT_FUNCTION(com_err_initialize))
         return 0;
     k5_mutex_lock(&et_list_lock);
 
     /* Remove the entry that matches the error table instance. */
-    for (ep = &et_list; *ep; ep = &(*ep)->next) {
-        if ((*ep)->table == et) {
-            e = *ep;
-            *ep = e->next;
-            free(e);
-            k5_mutex_unlock(&et_list_lock);
-            return 0;
-        }
+    for (epp = NULL, ep = et_list; ep && ep->table != et; epp = ep, ep = ep->next) ;
```

Comment:
Style here is:
- check values against NULL explicitly
- limit lines to 79 or fewer characters
- no space before semicolon.

Reply:
code test for null is unchanged against the original code